Use `shared` report_service and test factories

Swatinem opened the pull request:
### Purpose/Motivation

This switches most of the code using the `report_builder` service to use that same code imported from `shared` (except some functions specific to `api`).
    
Similarly, the `core.tests.factories` were also duplicated in shared, so this switches over to those factories, deleting them from `api`.

The primary reason here is that the `shared` code has been changed to remove the obsolete `report_builder` feature.
This feature was recently removed from `worker` and its configuration from the `k8s` config.
It is currently not enabled in production.

I believe this feature does not pull its weight, as querying multiple tables *and* fetching the `files_array` file from storage is most likely slower than just fetching the `report_json` from storage instead.

### Links to relevant tickets

https://github.com/codecov/engineering-team/issues/2554

matt-codecov commented on core/tests/factories.py:
```diff
@@ -66,7 +66,51 @@ class Meta:
+                        "t": [3, 20, 17, 3, 0, "82.00000", 0, 0, 0, 0, 0, 0, 0],
+                        "st": "uploaded",
+                        "se": None,
+                        # "st": "carriedforward",
+                        # "se": {
+                        #     "carriedforward_from": "56e05fced214c44a37759efa2dfc25a65d8ae98d"
+                        # },
```
?

matt-codecov commented on services/report.py:
```diff
@@ -61,42 +51,20 @@ def build_report(chunks, files, sessions, totals, report_class=None):
 
 @sentry_sdk.trace
-def build_report_from_commit(commit: Commit, report_class=None):
+def build_report_from_commit(commit: Commit, report_class=None) -> Report | None:
```
wait why aren't we just using `shared.reports.api_report_service`'s copy of this (which should maybe be renamed `shared.reports.django_report_service`)? is that an @adrian-codecov question?

Swatinem replied:
That is a good question, and I believe reducing the code duplication by removing one copy completely in favor of another is a good idea.

When working of this, I was bashing my head against the table quite a bit, I was even thinking that the definitions for `shared.core` were somehow shadowing `core`, even though I have no idea how that would be even possible.

This is good feedback, and I might just go ahead and remove this from api completely in favor of the definition in `shared`.

nora-codecov commented on requirements.in:
```diff
@@ -20,7 +20,7 @@ factory-boy
 freezegun
 https://github.com/codecov/opentelem-python/archive/refs/tags/v0.0.4a1.tar.gz#egg=codecovopentelem
-https://github.com/codecov/shared/archive/88117b96a4b420d88549b8df2649c3eb9c61c2a5.tar.gz#egg=shared
+https://github.com/codecov/shared/archive/bf3b6ee9973e7ec3b1eb57b6dce6ce96a18cbaa1.tar.gz#egg=shared
```
might as well grab the latest?

nora-codecov reviewed (approved):
love this cleanup, only question is about `OwnerFactory` - I see some spots where you've swapped it to the `shared` factory, and others where you've left it referring to the `api` factory, any reason other than this pr getting too big?

Swatinem commented:
> I see some spots where you've swapped it to the `shared` factory, and others where you've left it referring to the `api` factory

This might just be an oversight.
This PR still has a ton of failures, and merge conflicts by now, so I will turn it into a draft in the meantime before fixing all that. Thanks for the review nevertheless :-)

Swatinem commented:
I rebased the PR, and fixed all of the remaining tests, so things are green 🎉 
I would appreciate another review, and hope to get this landed rather sooner than later :-)
